dashboard: make repro logs public

xairy opened the pull request:
PR #4837 added displaying repro logs to syzbot but didn't make the logs publicly-accessible. Fix that.

xairy commented:
CI fails with:

```
    repro_test.go:401: expected OK, got error: 404: <!doctype html>
        <html>
        <head>
        	
        	<link rel="stylesheet" href="/static/style.css"/>
        	<script src="/static/common.js"></script>
        	
        
        	<title>syzbot</title>
        </head>
        <body>
        	resource not found: checkCrashTextAccess: found 0 crashes for ReproLog=5910974510923776
        	<br>
        	<a href="javascript:history.back()">back</a>
        </body>
        </html>
```

But I don't understand why. @a-nogikh, could you PTAL?

a-nogikh commented:
The problem is that we reference `textReproLog` not only from `Crash.ReproLog`, but also from `Bug.ReproAttempts.Log` (which is actually used by the failing `TestFailedReproLogs` test).

If it were apparent from the very beginning, we could have used different access policy rules for `textReproLog` and e.g. `textFailedReproLog`, but now we already have some collection of references from `Bug.ReproAttempts.TestFailedRepro.Log`, so we have to deal with it (*).

I think it's easiest now to introduce some new `checkBugTextAccess()` helper method and call both  `checkBugTextAccess()` and  `checkCrashTextAccess()` for `textReproLog` in `checkTextAccess()`.

***

(*) We currently don't display the failed repro attempts logs publicly, but probably we can start doing that? There's nothing special in there.

xairy commented:
Ah, I see, thanks!

I don't think it makes sense to expose failed repro attempts. At least not on the syzbot dashboard, this will only clutter it.

So I fixed the issue by falling back to `checkAccessLevel` for failed attempts, PTAL.

I also updated `access_test.go` to include `ReproLog`. However,  I noticed that the test succeeds regardless of the checks in `checkTextAccess`. No only for `ReproLog`, but for other texts too. I don't know if this is expected.

a-nogikh commented:
Thank you!
 
 > No only for ReproLog, but for other texts too. 
 
 Could you please share what exactly I should change locally to reproduce it?

xairy commented:
> Could you please share what exactly I should change locally to reproduce it?

For example, comment out the `textReproSyz` case in `checkTextAccess` and run the `TestAccess` test. It still succeeds.

a-nogikh commented:
That was probably due to https://github.com/google/syzkaller/pull/5018

xairy commented:
Ah, the non-short mode of running the tests makes the `TestAccess` test fail (with the `textReproSyz` case commented out). The short mode still succeeds, but I suppose that's OK.

Thank you!
